respy/simulate.py

```diff
@@ -543,6 +550,25 @@ def _apply_law_of_motion(df, optim_paras):
     df = df.copy()
     n_lagged_choices = optim_paras["n_lagged_choices"]
 
+    # Exogenous processes first, because the indexer requires the original states.
+    if transition_probabilities:
+        # Get indices which connect states in the state space and simulated agents.
+        # Subtract the minimum of indices (excluding invalid indices) because wages,
+        # etc. contain only wages in this period and normal indices select rows from all
+        # wages.
+        columns = create_core_state_space_columns(optim_paras)
+        indices = indexer[tuple(df[col].astype("int64") for col in columns)]
+
+        # Sample characteristics affected by exogenous processes.
+        dense_indices = list(transition_probabilities)
```

Comment:
Do we have to call them dense here the way that I understood it was that we do only have exogenous processes in that list?

Reply:
No, you sample the whole dense vector although only the entries of the exogenous process change. It is easier to deal with if you do not need to rebuild the dense indices. But, if we agree to undo the melting of the probabilities, this is likely to change as well.